src/api/mod.rs

```diff
@@ -96,3 +97,38 @@ pub(in crate::api) async fn check_community_ban(
     Ok(())
   }
 }
+
+pub(in crate::api) async fn linked_instances(pool: &DbPool) -> Result<Vec<String>, LemmyError> {
+  let mut instances: Vec<String> = Vec::new();
+
+  if Settings::get().federation.enabled {
+    let distinct_communities = blocking(pool, move |conn| {
+      Community::distinct_federated_communities(conn)
+    })
+    .await??;
+
+    for actor_id in &distinct_communities {
+      instances.push(Url::parse(actor_id)?.host_str().unwrap_or("").to_string());
+    }
+
+    let mut allowed_instances = Settings::get().get_allowed_instances();
+    instances.append(&mut allowed_instances);
+
+    let blocked_instances = Settings::get().get_blocked_instances();
+    for blocked in &blocked_instances {
+      instances.retain(|i| !i.eq(blocked));
+    }
+
+    // Sort and remove dupes
+    instances.sort_unstable();
+    instances.dedup();
+
+    // Remove the empties
+    instances.retain(|d| !d.eq(""));
+
+    // Remove the current instance
+    instances.retain(|d| !d.eq(&Settings::get().hostname));
```

Comment:
I think this function could be simplified by using stream operators, like filter() instead of retain(), and map() to get the actor IDs.

Reply:
One thing performance-wise, is that it seems like I should be using a Set: https://stackoverflow.com/questions/64019451/how-do-i-remove-the-elements-of-vector-that-occur-in-another-vector-in-rust?noredirect=1#comment113207400_64019451

I'll see if I can convert some of these to stream operators.